src/ssl/PeekingPeerConnector.cc

```diff
@@ -166,7 +166,7 @@ Ssl::PeekingPeerConnector::initialize(Security::SessionPointer &serverSession)
         if (hostName)
             SSL_set_ex_data(serverSession.get(), ssl_ex_index_server, (void*)hostName);
 
-        Must(!csd->serverBump() || csd->serverBump()->step <= Ssl::bumpStep2);
+        Must(!csd->serverBump() || csd->serverBump()->step <= xaStepTlsBump2);
```

Comment:
Since we are adding other steps, I feel uncomfortable with doing open-ended comparison here (like `<= ...2` does). Since this pattern happens more than once, we should add a helper. We could do

```suggestion
        Must(!csd->serverBump() || csd->serverBump()->atOrBefore(xaStepTlsBump2);
```

but I think we should do this instead because it is more readable and will cover more use cases, including `>`.

```suggestion
        Must(!csd->serverBump() || csd->serverBump()->at(xaStepTlsBump1, xaStepTlsBump2);
```

Reply:
I used various ways to implement at() method, initializers list, default set for second argument and finally I just overloaded the at() method providing two variants. 

git commit 0719be7e27

Reply:
I think adding a two-step at() is fine in this context because we only have three possible steps and nobody should be calling at() with all the steps. If you are curious, here is one of the simplest implementation I could come up with that could handle more steps:

```C++
    /// whether we are at one of the given SslBump steps
    template <typename... XactionSteps>
    bool at(const XactionStep one, const XactionSteps... other) const { return (step == one) || at(other...); }
    bool at() const { return false; }
```

Again, I think your design is better for our three-step use case. I am fixing the implementation but will leave your design. Thus, I am resolving this change request.